tests/integration/istioio/trafficmanagement/scripts/traffic_shifting.txt

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# setup bookinfo & sleep pods
+kubectl label namespace default istio-injection=enabled --overwrite || true
+
+kubectl apply -f samples/bookinfo/platform/kube/bookinfo.yaml
+kubectl apply -f samples/bookinfo/networking/bookinfo-gateway.yaml
+kubectl apply -f samples/bookinfo/networking/destination-rule-all.yaml
+kubectl apply -f samples/sleep/sleep.yaml
+
+for deploy in "productpage-v1" "details-v1" "ratings-v1" "reviews-v1" "reviews-v2" "reviews-v3" "sleep"; do
+  if ! kubectl rollout status deployment "$deploy" --timeout 5m; then
```

Comment:
kubectl wait may work better here?

Reply:
I'm not I'm not actually sure , I've never used status

Reply:
I saw it used in another test (authz_http), I don't know which is better, should I change this to `kubectl wait`?

Reply:
I probably wrote the code you're referring to :) ... I seem to remember kubectl wait not doing exactly what I wanted. But it may have been due to my lack of understanding of the command.

This sort of thing is going to be very common in our tests. We should add a wait function to the framework to help automate this.